text/0088-klt-instances.md

```diff
@@ -0,0 +1,102 @@
+# KEP 88: Adding Instances to KeptnApp 
+
+**State: DRAFTING**
+
+## Motivation
+At the moment, the lifecycle toolkit and its resources are operating in a single instance mode, meaning that the lifecycle toolkit is only able to manage a single instance of a workload and no distinction is made between multiple instances (stages/environments) of the same workload. 
+
+This KEP describes the changes required to enable the lifecycle toolkit to manage multiple instances of a workload / application.
+
+## Current Design
+Currently, the Keptn Lifecycle Toolkit (Controller) is managing a single instance of an application or workload under the assumption that it is running in a single namespace and if multiple instances exist, there is some distinction between them also in the observability solution.
+
+Therefore, there are no explicit references to the instance of the application in the lifecycle toolkit resources.
+
+
+## Assumptions / Definitions
+- In the Keptn Lifecycle Toolkit, an instance describes a single instance of a workload or application. This can be a single stage or environment of a workload or application.
+- An instance has to be specified on the workload and application level
+- If an instance is not specified, the lifecycle toolkit will assume that there is only a single instance of the workload or application
+- If an instance is not specified on the workload level, the lifecycle toolkit will lookup the corresponding application and use the instance specified there
+- Properties of an instance are defined in the `KeptnApp` resource
+- Promotion information can be specified in the Instance, there can be successors, but no predecessors of an instance
+- This KEP does not cover the promotion of an instance, this will be covered in a separate KEP / Issue 
+
+## Proposed Design
+To add the ability to manage multiple instances of an application, the following structure will be added to the `KeptnApp` resource:
+
+```yaml
+apiVersion: keptn.sh/v1alpha3
+kind: KeptnApp
+metadata:
+  name: podtatohead
+  namespace: keptn
+spec:
+  instance: <name of the current instance>
+  [...]
+  instances:
+    defaults:
+      repository: <git repository>
+      branch: <git branch>
+      path: <path to the manifest>
+      reposecret: <secret to access the repository>
```

Comment:
TC discussion: here we could use `secretName` and `secretKey` to fetch a K8s secret.